tools/api-performance-test/performance.py

```diff
@@ -0,0 +1,333 @@
+  for filename in os.listdir(directory_path):
+    if filename.endswith('.json'):
+      file_path = os.path.join(directory_path, filename)
+      read_from_json(file_path, ecosystem_map, bug_map, project_map)
+  return ecosystem_map, bug_map, project_map
+
+
+async def make_http_request(session, request_url, request_type, request_body):
+  """Makes one HTTP request"""
+  try:
+    timeout = aiohttp.ClientTimeout(total=None, sock_connect=80, sock_read=80)
+    if request_type == 'GET':
+      async with session.get(request_url):
+        pass  # We're not awaiting the response, just sending the request
+    elif request_type == 'POST':
+      async with session.post(request_url, json=request_body, timeout=timeout):
+        # print(f'request: {request_body}, response: {response.status}')
+        pass  # We're not awaiting the response, just sending the request
+  except Exception as e:
+    print(f'Error sending request {request_url} with body'
+          f'{request_body}: {type(e)}')
+
+
+async def make_http_requests_async(request_ids, bug_map, url, batch_size,
+                                   payload_func):
+  """Makes the required number of HTTP requests per second."""
+  begin_time = time.monotonic()
+  print(f'[{begin_time}] Running make request {payload_func.__name__} '
+        f'for {TOTAL_RUNTIME} seconds')
+
+  total_run_time = time.monotonic() - begin_time
+  index = 0
+  length = len(request_ids)
+  async with aiohttp.ClientSession() as session:
+    while total_run_time < TOTAL_RUNTIME:
+      start_time = time.monotonic()
+
+      batch_request_ids = request_ids[index:batch_size + index]
+      if payload_func.__name__ == vulnerability_payload.__name__:
+        for request_id in batch_request_ids:
+          asyncio.create_task(
+              make_http_request(session, f'{url}/{request_id}', 'GET',
+                                payload_func()))
+      elif payload_func.__name__ == batch_payload.__name__:
+        for _ in range(0, batch_size):
+          asyncio.create_task(
+              make_http_request(session, url, 'POST',
+                                payload_func(request_ids, bug_map)))
+      else:
+        for request_id in batch_request_ids:
+          asyncio.create_task(
+              make_http_request(session, url, 'POST',
+                                payload_func(request_id, bug_map)))
+      index += batch_size
+      if index >= length:
+        index = 0
+
+      end_time = time.monotonic()
+      time_elapsed = end_time - start_time
+      if time_elapsed < 1:
+        await asyncio.sleep(1 - time_elapsed)
+      total_run_time = time.monotonic() - begin_time
+
+
+def package_payload(request_id, bug_map):
+  """package/project query payload"""
+  package = random.choice(bug_map[request_id].projects)
+  ecosystem = random.choice(bug_map[request_id].ecosystems)
+  return {"package": {"name": package, "ecosystem": ecosystem}}
+
+
+def version_payload(request_id, bug_map):
```

Comment:
nit: Rename these payload generator functions to `build_version_payload` to more easily differentiate them from just a variable of the payload itself.